### PR TITLE
Add properties to Domain cs proj

### PR DESCRIPTION
**Pull request by Korgath**
## Description

Added the following properties to Domain CSProj:

- LangVersion ->9.0
- Nullable ->enable
- TreatWarningsAsErrors ->true
- RunAnalyzersDuringBuild ->true
- GenerateDocumentationFile ->true

Fixed errors coming from this change

## References

* #0000

### CODE REVIEW

**Review by erduna — APPROVED** [review]

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/MeteringPoints/Rules/ChangeEnergySupplier/MoveOutRegisteredOnSameDateIsNotAllowedRule.cs [hunk 1]
```diff
@@ -38,8 +38,8 @@ internal class MoveOutRegisteredOnSameDateIsNotAllowedRule : IBusinessRule
...
         //                                    .Equals(_effectuationDate.ToDateTimeUtc().Date));
         // }
-        public bool IsBroken => false;
-
-        public ValidationError Error => new MoveOutRegisteredOnSameDateIsNotAllowedRuleError();
-    }
+    //     public bool IsBroken => false;
+    //
+    //     public ValidationError ValidationError => new MoveOutRegisteredOnSameDateIsNotAllowedRuleError();
+    // }
```
Comment (cmgrd) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/MeteringPoints/Rules/ChangeEnergySupplier/MoveOutRegisteredOnSameDateIsNotAllowedRule.cs:44]: Commented out code

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/SeedWork/IBusinessRule.cs [hunk 2]
```diff
@@ -27,6 +27,6 @@ public interface IBusinessRule
         /// <summary>
         /// Validation details
         /// </summary>
-        ValidationError Error { get; }
+        ValidationError ValidationError { get; }
```
Comment (cmgrd) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/SeedWork/IBusinessRule.cs:30]: Why the name change?

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/Consumer.cs [hunk 3]
```diff
@@ -27,15 +27,15 @@ public Consumer(ConsumerId consumerId, CprNumber cprNumber, ConsumerName name)
...
+            AddDomainEvent(new ConsumerCreated(ConsumerId.Value, CprNumber.Value, CvrNumber?.Value, _name.FullName));
         }
 
         public Consumer(ConsumerId consumerId, CvrNumber cvrNumber, ConsumerName name)
             : this(consumerId, name)
         {
             ConsumerId = consumerId ?? throw new ArgumentNullException(nameof(consumerId));
             CvrNumber = cvrNumber ?? throw new ArgumentNullException(nameof(cvrNumber));
-            AddDomainEvent(new ConsumerCreated(ConsumerId.Value, CprNumber?.Value, CvrNumber?.Value, _name.FullName));
+            AddDomainEvent(new ConsumerCreated(ConsumerId.Value, CprNumber?.Value, CvrNumber.Value, _name.FullName));
```
Comment (cmgrd) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/Consumer.cs:38]: If the not-used property will be null, shouldn't we just pass null then? It cannot be both, right?

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/SeedWork/BusinessRulesValidationResult.cs [hunk 4]
```diff
@@ -13,26 +13,28 @@
...
-        public BusinessRulesValidationResult(List<IBusinessRule> rules)
+        public BusinessRulesValidationResult(IEnumerable<IBusinessRule> rules)
         {
             SetValidationErrors(rules);
         }
 
         public bool Success => Errors.Count == 0;
 
-        public List<ValidationError> Errors { get; private set; } = new();
+        public ICollection<ValidationError> Errors { get; private set; } = new List<ValidationError>();
```
Comment (cmgrd) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/SeedWork/BusinessRulesValidationResult.cs:31]: Why do we prefer a collection over a list?

**Review by cmgrd — CHANGES_REQUESTED** [review]

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/CprNumber.cs [hunk 5]
```diff
@@ -41,7 +41,7 @@ public static CprNumber Create(string cprValue)
 
         public static BusinessRulesValidationResult CheckRules(string? cprValue)
         {
-            return new BusinessRulesValidationResult(new List<IBusinessRule>()
+            return new BusinessRulesValidationResult(new Collection<IBusinessRule>()
```
Comment (kft) [inline source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/CprNumber.cs:44]: Why change to collection ?

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/Consumer.cs (hunk 3 above)
Comment (Korgath) [reply source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/Consumer.cs:38]: Correct

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/CprNumber.cs (hunk 5 above)
Comment (Korgath) [reply source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/Consumers/CprNumber.cs:44]: Lots of stylecop complaining implementing the properties - forgot to set this back to List.

File: source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/MeteringPoints/Rules/ChangeEnergySupplier/MoveOutRegisteredOnSameDateIsNotAllowedRule.cs (hunk 1 above)
Comment (Korgath) [reply source/business-workflow/source/Energinet.DataHub.MarketRoles.Domain/MeteringPoints/Rules/ChangeEnergySupplier/MoveOutRegisteredOnSameDateIsNotAllowedRule.cs:44]: Yes, Part of the TODO. Just commented the last part out to make it build

**Review by erduna — APPROVED** [review]

**Review by cmgrd — APPROVED** [review]

**Pull request merged**